test/parallel/test-timers-interval-promisified.js

```diff
@@ -246,3 +246,14 @@ process.on('multipleResolves', common.mustNotCall());
     setPromiseTimeout(time_unit * 3).then(() => post = true),
   ]).then(common.mustCall());
 }
+
+(async () => {
+  const signal = AbortSignal.abort('boom');
+  try {
+    const iterable = timerPromises.setInterval(2, undefined, { signal });
+    // eslint-disable-next-line no-unused-vars
+    for await (const _ of iterable) {}
+  } catch (err) {
+    assert.strictEqual(err.cause, 'boom');
+  }
```

Comment:
```suggestion
    await assert.rejects(
      () => {
        const iterable = timerPromises.setImmediate(undefined, { signal });
        // eslint-disable-next-line no-unused-vars
        for await (const _ of iterable) {}
      },
      {  cause: 'boom' }
    }
```

Reply:
I actually think `try`/`catch` is fine for this specific use case, clearer than the `assert.rejects` suggestion. For the other cases, I agree we should prefer using `asserts.rejects`.

Reply:
I might be wrong but I believe that could miss the catch part in case the `for await` would not throw without anyone noticing.

Reply:
I'll keep the try/catch on this one but it does need an `assert.fail` after the loop to catch it from falling through